File: trunk/theLink/example/cc/testserver.cc
/**
 *  \file       theLink/example/cc/testserver.cc
 *  \brief      \$Id$
 *  
 *  (C) 2009 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev$
 *  \author     EMail: aotto1968 at users.berlios.de
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */

#include "ccmsgque.h"

using namespace ccmsgque;

class testserver : public MqC, public IServerSetup {
  public:
    testserver(MqS *tmpl) : MqC(tmpl) {}
  private:
    void GTCX () {
      SendSTART();
      SendI(LinkGetCtxId());
      SendC("+");
      if (LinkIsParent()) {
	SendI(-1);
      } else {
	SendI(LinkGetParent()->LinkGetCtxId());
      }
      SendC("+");
      SendC(ConfigGetName());
      SendC(":");
      SendRETURN();
    }
    void ServerSetup() {
      ServiceCreate("GTCX", CallbackF(&testserver::GTCX));
    }
};

int MQ_CDECL main (int argc, MQ_CST argv[])
{
  testserver *ctx = MqFactoryC<testserver>::Add("testserver").New();
  try {
    ctx->LinkCreateVC(argc, argv);
    ctx->ProcessEvent (MQ_WAIT_FOREVER);
  } catch (const exception& e) {
    ctx->ErrorSet(e);
  }
  ctx->Exit ();
}

File: tags/NHI1-0.16/theLink/example/vb/testserver.vb
REM
REM  \file       theLink/example/vb/testserver.vb
REM  \brief      \$Id$
REM  
REM  (C) 2009 - NHI - #1 - Project - Group
REM  
REM  \version    \$Rev$
REM  \author     EMail: aotto1968 at users.berlios.de
REM  \attention  this software has GPL permissions to copy
REM              please contact AUTHORS for additional information
REM

Imports System
Imports csmsgque

Public Module example
  Private Class testserver
    Inherits MqS
    Implements IServerSetup
    
    ' constructor
    Public Sub New(ByVal tmpl As MqS)
      MyBase.New(tmpl)
    End Sub

    Private Sub GTCX()
      SendSTART()
      SendI(LinkGetCtxId())
      SendC("+")
      If LinkIsParent() Then
        SendI(-1)
      Else
        SendI(LinkGetParent().LinkGetCtxId())
      End If
      SendC("+")
      SendC(ConfigGetName())
      SendC(":")
      SendRETURN()
    End Sub

    Private Sub ServerSetup() Implements IServerSetup.ServerSetup
      ServiceCreate("GTCX", AddressOf GTCX)
    End Sub
  End Class

  Sub Main(ByVal args() As String)
    Dim srv As testserver = MqFactoryS(Of testserver).Add().[New]()
    Try
      srv.LinkCreate(args)
      srv.ProcessEvent(MqS.WAIT.FOREVER)
    Catch ex As Exception
      srv.ErrorSet(ex)
    Finally
      srv.Exit()
    End Try
  End Sub
End Module
